packages/gestalt-charts/src/ChartGraph/useTabularData.js

```diff
@@ -0,0 +1,139 @@
+// @flow strict-local
+import { useDefaultLabel } from 'gestalt';
+
+export type SortChangeType = 'series' | 'x' | 'y';
+export type FilterIdType = null | SortChangeType;
+export type FilterOrderType = 'desc' | 'asc';
+
+export type TransformedTabularDataType = $ReadOnlyArray<{|
+  series: string,
+  xAxis: number | string,
+  yAxis: number,
+|}>;
+
+type UseBuildCsvDataProps = ({|
+  transformedTabularData: TransformedTabularDataType,
+  isHorizontalLayout: boolean,
+|}) => string;
+
+export const useBuildCsvData: UseBuildCsvDataProps = ({
+  transformedTabularData,
+  isHorizontalLayout,
+}) => {
+  const { tableSeriesText, tableXAxisText, tableYAxisText } = useDefaultLabel('ChartGraph');
+
+  return `${tableSeriesText},${isHorizontalLayout ? tableXAxisText : tableYAxisText},${
+    isHorizontalLayout ? tableYAxisText : tableXAxisText
+  }\n${transformedTabularData
+    .map((x) => Object.values(x))
+    .map((e) => e.join(','))
+    .join('\n')}`;
```

Comment:
nit: Do you think it's a good idea to wrap it with `useMemo`? I am not sure how often this hook is used.

Reply:
It´s a good concern. I can try to memoize. I think this is only called on the TabularDataModal component so there shouldn't be many rerenders in general after the mounting of Modal. If it were within ChartGraph, I think there could be more unrelated rerenders
@diyorbek

Reply:
DONE